### PR TITLE
Fix malformed uname string

### LINKED ISSUE #2726: Initial uname block malformed
The recently added log line for uname info (#2709) generates malformed output:

```
time="2017-05-16T13:07:04Z" level=info msg="Host details (Linux\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\
x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 4.9.24-coreos\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0
0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 #1 SMP Wed Apr 26 21:44:23 UTC 2017\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0
0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 x86_64\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00
\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 prometheus-b-0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\
x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 (none)\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x
00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00)" source="main.go:90"
```

IIRC, the syscall provides a struct of `[63]int8` array, which are probably cstr, which have to be truncated correctly when converting them to strings.

### PR DESCRIPTION
**Pull request by CrazyByDefault**
Fixes #2726 

@fabxc

### CODE REVIEW

File: cmd/prometheus/uname_linux_int8.go [hunk 1]
```diff
@@ -16,10 +16,12 @@
 
 package main
 
+import "strings"
+
 func charsToString(ca []int8) string {
 	s := make([]byte, len(ca))
 	for i, c := range ca {
 		s[i] = byte(c)
```
Comment (Gouthamve) [inline cmd/prometheus/uname_linux_int8.go]: Can you break out of the loop if the byte is a null byte? Just print out the `int8` value and check against that.

File: cmd/prometheus/uname_linux_int8.go [hunk 2]
```diff
@@ -17,9 +17,14 @@
...
-	for i, c := range ca {
-		s[i] = byte(c)
+	s := make([]byte, 0, len(ca))
+	for _, c := range ca {
+
+		if byte(c) != 0 {
+			s = append(s, byte(c))
+		} else {
+			break
+		}
```
Comment (fabxc) [inline cmd/prometheus/uname_linux_int8.go]: Little Go tip, reduce the number of intended block to make the code simpler

```
if byte(c) == 0 {
    break
}
s = append(s, byte(c))
```

File: cmd/prometheus/uname_linux_int8.go [hunk 3]
```diff
@@ -17,9 +17,14 @@
...
 	}
-	return string(s[0:len(ca)])
+	return string(s[:])
```
Comment (fabxc) [inline cmd/prometheus/uname_linux_int8.go]: Just `return string(s)` is enough here. `[:]` takes a slice of the given array or slice, but as `s` is the slice we want already, that's not necessary.

Comment (CrazyByDefault) [inline cmd/prometheus/uname_linux_int8.go]: Oh! Thanks, thought slices required the `[:]` notation whenever mentioned

File: cmd/prometheus/uname_linux_int8.go (hunk 2 above)
Comment (CrazyByDefault) [inline cmd/prometheus/uname_linux_int8.go]: Yeah no idea why I didn't do this in the first place

**Comment by fabxc** [comment]
Thanks 👍

**Pull request merged**